src/styles/main.scss

```diff
@@ -0,0 +1,35 @@
+.main {
+  justify-content: center;
+  margin: 0;
+  padding: 50px 40px;
+  display: grid;
+
+  --column-count: 4;
+
+  grid-template-columns: repeat(var(--column-count), 200px);
+  gap: 46px 48px;
+}
+
+@media (min-width: 300px) {
+  .main {
+    --column-count: 1;
+  }
+}
+
+@media (min-width: 488px) {
+  .main {
+    --column-count: 2;
+  }
+}
+
+@media (min-width: 768px) {
+  .main {
+    --column-count: 3;
+  }
+}
+
+@media (min-width: 1024px) {
+  .main {
+    --column-count: 4;
+  }
+}
```

Comment:
The media queries are correctly implemented to adjust the '--column-count' variable based on screen width, ensuring the grid layout adapts to different screen sizes as required.